From d20e6239bd0b5a25b7eba19c3e5c0de44f8f44dd Mon Sep 17 00:00:00 2001
From: Ahmad Samir <a.samirh78@gmail.com>
Date: Wed, 8 Oct 2025 01:01:11 +0300
Subject: [PATCH 418/553] tests/auto/gui: port away from QPair

Task-number: QTBUG-115841
Pick-to: 6.8 6.5
Change-Id: If0a462433a283189ad975f0ca92afea69234f70e
Reviewed-by: Ivan Solovev <ivan.solovev@qt.io>
(cherry picked from commit c69279a4d59256f2b610fb90871e979a2967a953)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 .../qsurfaceformat/tst_qsurfaceformat.cpp     |  4 +--
 .../gui/painting/qpainter/tst_qpainter.cpp    | 31 ++++++++-----------
 tests/auto/gui/rhi/qrhi/tst_qrhi.cpp          |  2 +-
 tests/auto/gui/rhi/qshader/tst_qshader.cpp    |  5 +--
 4 files changed, 19 insertions(+), 23 deletions(-)

diff --git a/tests/auto/gui/kernel/qsurfaceformat/tst_qsurfaceformat.cpp b/tests/auto/gui/kernel/qsurfaceformat/tst_qsurfaceformat.cpp
index 3f655bd905d..28085e1405f 100644
--- a/tests/auto/gui/kernel/qsurfaceformat/tst_qsurfaceformat.cpp
+++ b/tests/auto/gui/kernel/qsurfaceformat/tst_qsurfaceformat.cpp
@@ -54,10 +54,10 @@ void tst_QSurfaceFormat::versionCheck()
     format.setMinorVersion(formatMinor);
     format.setMajorVersion(formatMajor);
 
-    QCOMPARE(format.version() >= qMakePair(compareMajor, compareMinor), expected);
+    QCOMPARE(format.version() >= std::pair(compareMajor, compareMinor), expected);
 
     format.setVersion(formatMajor, formatMinor);
-    QCOMPARE(format.version() >= qMakePair(compareMajor, compareMinor), expected);
+    QCOMPARE(format.version() >= std::pair(compareMajor, compareMinor), expected);
 }
 
 #include <tst_qsurfaceformat.moc>
diff --git a/tests/auto/gui/painting/qpainter/tst_qpainter.cpp b/tests/auto/gui/painting/qpainter/tst_qpainter.cpp
index bd59ee1e1b2..f0802e0150f 100644
--- a/tests/auto/gui/painting/qpainter/tst_qpainter.cpp
+++ b/tests/auto/gui/painting/qpainter/tst_qpainter.cpp
@@ -3842,26 +3842,23 @@ void tst_QPainter::linearGradientSymmetry_data()
     QTest::addColumn<QGradientStops>("stops");
 
     if (sizeof(qreal) != sizeof(float)) {
-        QGradientStops stops;
-        stops << qMakePair(qreal(0.0), QColor(Qt::blue));
-        stops << qMakePair(qreal(0.2), QColor(220, 220, 220, 0));
-        stops << qMakePair(qreal(0.6), QColor(Qt::red));
-        stops << qMakePair(qreal(0.9), QColor(220, 220, 220, 255));
-        stops << qMakePair(qreal(1.0), QColor(Qt::black));
+        QGradientStops stops = {{qreal(0.0), QColor(Qt::blue)},
+                                {qreal(0.2), QColor(220, 220, 220, 0)},
+                                {qreal(0.6), QColor(Qt::red)},
+                                {qreal(0.9), QColor(220, 220, 220, 255)},
+                                {qreal(1.0), QColor(Qt::black)}};
         QTest::newRow("multiple stops") << stops;
     }
 
     {
-        QGradientStops stops;
-        stops << qMakePair(qreal(0.0), QColor(Qt::blue));
-        stops << qMakePair(qreal(1.0), QColor(Qt::black));
+        QGradientStops stops = {{qreal(0.0), QColor(Qt::blue)},
+                                {qreal(1.0), QColor(Qt::black)}};
         QTest::newRow("two stops") << stops;
     }
 
     if (sizeof(qreal) != sizeof(float)) {
-        QGradientStops stops;
-        stops << qMakePair(qreal(0.3), QColor(Qt::blue));
-        stops << qMakePair(qreal(0.6), QColor(Qt::black));
+        QGradientStops stops = {{qreal(0.3), QColor(Qt::blue)},
+                                {qreal(0.6), QColor(Qt::black)}};
         QTest::newRow("two stops 2") << stops;
     }
 }
@@ -3912,12 +3909,10 @@ void tst_QPainter::gradientPixelFormat()
     QImage a(8, 64, QImage::Format_ARGB32_Premultiplied);
     QImage b(8, 64, format);
 
-
-    QGradientStops stops;
-    stops << qMakePair(qreal(0.0), QColor(Qt::blue));
-    stops << qMakePair(qreal(0.3), QColor(Qt::red));
-    stops << qMakePair(qreal(0.6), QColor(Qt::green));
-    stops << qMakePair(qreal(1.0), QColor(Qt::black));
+    QGradientStops stops = {{qreal(0.0), QColor(Qt::blue)},
+                            {qreal(0.3), QColor(Qt::red)},
+                            {qreal(0.6), QColor(Qt::green)},
+                            {qreal(1.0), QColor(Qt::black)}};
 
     a.fill(0);
     b.fill(0);
diff --git a/tests/auto/gui/rhi/qrhi/tst_qrhi.cpp b/tests/auto/gui/rhi/qrhi/tst_qrhi.cpp
index ab0e8693644..511007e0ae4 100644
--- a/tests/auto/gui/rhi/qrhi/tst_qrhi.cpp
+++ b/tests/auto/gui/rhi/qrhi/tst_qrhi.cpp
@@ -3525,7 +3525,7 @@ void tst_QRhi::renderToTextureMultipleUniformBuffersAndDynamicOffset()
     // "see" an all zero matrix and zero opacity, thus leading to different
     // rendering output. This way we can verify if using dynamic offsets, and
     // more than one at the same time, is functional.
-    QVarLengthArray<QPair<int, quint32>, 2> dynamicOffset = {
+    QVarLengthArray<std::pair<int, quint32>, 2> dynamicOffset = {
         { 0, quint32(ubufElemSize * 2) },
         { 1, quint32(ubuf2ElemSize * 3) },
     };
diff --git a/tests/auto/gui/rhi/qshader/tst_qshader.cpp b/tests/auto/gui/rhi/qshader/tst_qshader.cpp
index 9e179c95c35..371da0d800c 100644
--- a/tests/auto/gui/rhi/qshader/tst_qshader.cpp
+++ b/tests/auto/gui/rhi/qshader/tst_qshader.cpp
@@ -279,7 +279,7 @@ void tst_QShader::mslResourceMapping()
 
     QCOMPARE(resMap.size(), 2);
     QCOMPARE(resMap.value(0).first, 0); // mapped to native buffer index 0
-    QCOMPARE(resMap.value(1), qMakePair(0, 0)); // mapped to native texture index 0 and sampler index 0
+    QCOMPARE(resMap.value(1), std::pair(0, 0)); // mapped to native texture index 0 and sampler index 0
 }
 
 void tst_QShader::serializeShaderDesc()
@@ -667,7 +667,8 @@ void tst_QShader::loadV7()
     QCOMPARE(tese.description().inputBuiltinVariables()[3].type, QShaderDescription::TessCoordBuiltin);
 
     QCOMPARE(tese.nativeResourceBindingMap(QShaderKey(QShader::MslShader, QShaderVersion(12))).size(), 1);
-    QCOMPARE(tese.nativeResourceBindingMap(QShaderKey(QShader::MslShader, QShaderVersion(12))).value(0), qMakePair(0, -1));
+    QCOMPARE(tese.nativeResourceBindingMap(QShaderKey(QShader::MslShader, QShaderVersion(12))).value(0),
+             std::pair(0, -1));
 
     QShader frag = getShader(QLatin1String(":/data/metal_enabled_tessellation_v7.frag.qsb"));
     QVERIFY(frag.isValid());
-- 
2.51.2

